package/MDAnalysis/analysis/encore/clustering/ClusteringMethod.py

```diff
@@ -195,6 +195,13 @@ def __init__(self,
                 (default is 50). Parameter in the Affinity Propagation for
                 clustering.
 
+            random_state : int, RandomState instance or None, optional
+                Used to control the starting state(default is 0). If an
+                integer is given, it fixes the seed for the new random number
+                generator. Use of integer is recommended to mimic the function
+                call. None uses the global random state instance from
```

Comment:
Given that we already pass ``kwargs`` into AffinityProgation, I can only see giving a default to `random_state` causing trouble for us in the future. In scikit-learn 0.23, the default setting "warn" warns us that the default will change from "warn" to `None` in 1.0. 

https://github.com/scikit-learn/scikit-learn/blob/95119c13af77c76e150b753485c662b7c52a41a2/sklearn/cluster/_affinity_propagation.py#L147

Instead, it would be more useful to document `**kwargs`.

Reply:
If we go with the above approach and include the `random_state` parameter in `**kwargs`, and do not provide any `random_state` default value. We can only solve this part of the issue by providing a constant value to `random_state` parameter for the `sklearn.cluster.AffinityPropagation()` function call.

Reply:
@thadanipaarth I elaborated above, but as you can tell by the length of the comment, the value of `random_state` is a somewhat important choice. Whichever way we choose to go on the default value, I think the documentation will have to be more specific. For example: "starting state" of what? Is the random number generator "new", or has scikit-learn just always hardcoded the seed? Would the use of *any* integer mimic the historical performance of Encore, or just 0? And which function call does the value of 0 mimic? These are all questions that users may have, and that we should be able to answer or at least link to the scikit-learn documentation for more information.

Reply:
Yes, I completely got the point. I will update the documentation in detail once we decide on the final approach.